runtime/src/main/java/com/dylibso/chicory/runtime/Machine.java

```diff
@@ -1489,12 +1509,26 @@ void eval(List<Instruction> code) throws ChicoryException {
                             if (Double.isNaN(tos)) {
                                 throw new WASMRuntimeException("invalid conversion to integer");
                             }
-                            var tosL = (long) tos;
-                            if (tos == (double) Long.MAX_VALUE) {
```

Comment:
@thomasdarimont I'm interested to get your feedback on removing this check - this is the one bit I didn't understand, though I don't think we need it with these considerations of wrap-around added in.

My expectation is that there are probably faster & cleaner ways to handle this truncation with some bit-twiddling, but for the moment this gets the tests passing.

Reply:
I had to fiddle quite a bit with the bitwise representations to get the tests pass. I'm fine with any more elegant or straight forward solution:)